resourceCloudflareLoadBalancerStateUpgradeV1: panic: interface {} is nil, not []interface {}

### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

```
Terraform v1.0.9
on linux_amd64
+ provider registry.terraform.io/aminueza/minio v1.2.0
+ provider registry.terraform.io/brinkmanlab/galaxy v0.3.0
+ provider registry.terraform.io/cloudflare/cloudflare v3.2.0
+ provider registry.terraform.io/hashicorp/aws v3.63.0
+ provider registry.terraform.io/kreuzwerker/docker v2.15.0
+ provider registry.terraform.io/louy/uptimerobot v0.5.1
+ provider registry.terraform.io/terraform-provider-openstack/openstack v1.44.0
```

### Affected resource(s)

cloudflare_load_balancer
plugin core

### Terraform configuration files

```hcl
resource "cloudflare_load_balancer" "sfu" {
  zone_id = cloudflare_zone.brinkmanlab_ca.id
  default_pool_ids = [
    data.terraform_remote_state.core.outputs.cloudflare_load_balancer_pool_sfu.id,
  ]
  fallback_pool_id = data.terraform_remote_state.core.outputs.cloudflare_load_balancer_pool_fallback.id
  name             = "lb.sfu.${local.domain}"
  proxied          = false
  steering_policy  = "off"
}
```


### Debug output

if this is needed I will provide it via email. I can't be sure where any secrets or tokens are.

### Panic output

```
Stack trace from the terraform-provider-cloudflare_v3.2.0 plugin:

panic: interface conversion: interface {} is nil, not []interface {}

goroutine 275 [running]:
github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareLoadBalancerStateUpgradeV1({0xc0000d3a40, 0x506365}, 0xc00017e480, {0xe36800, 0xc0005542d0})
        github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_load_balancer_migrate.go:56 +0x29d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).upgradeJSONState(0xc0001306a8, {0x120bce8, 0xc0007b37c0}, 0xe36800, 0xc0005542d0, 0xc000995ab8)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.8.0/helper/schema/grpc_provider.go:424 +0x97
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).UpgradeResourceState(0xc0001306a8, {0x120bce8, 0xc0007b37c0}, 0xc0006302e0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.8.0/helper/schema/grpc_provider.go:300 +0x365
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).UpgradeResourceState(0xc0005179e0, {0x120bd90, 0xc000befe60}, 0xc0006165a0)
        github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/tf5server/server.go:281 +0x130
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_UpgradeResourceState_Handler({0xfa26e0, 0xc0005179e0}, {0x120bd90, 0xc000befe60}, 0xc00078cea0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:308 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000af500, {0x121d688, 0xc0005ee000}, 0xc000b05440, 0xc0002fa1b0, 0x18b8e00, 0x0)
        google.golang.org/grpc@v1.40.0/server.go:1297 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc0000af500, {0x121d688, 0xc0005ee000}, 0xc000b05440, 0x0)
        google.golang.org/grpc@v1.40.0/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.40.0/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.40.0/server.go:939 +0x294

Error: The terraform-provider-cloudflare_v3.2.0 plugin crashed!
```

### Expected output

Update successfully

### Actual output

```
│ Error: Plugin did not respond
│ 
│   with cloudflare_load_balancer.sfu,
│   on dns.tf line 105, in resource "cloudflare_load_balancer" "sfu":
│  105: resource "cloudflare_load_balancer" "sfu" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).UpgradeResourceState call. The plugin logs may contain more details.

```

### Steps to reproduce

1. Upgrade to v3.2.0 of cloudflare plugin
1. terraform apply

### Additional factoids

_No response_

### References

https://github.com/cloudflare/terraform-provider-cloudflare/blob/6f95e1b7449fb06b09a3f8bbd4e88b0cb1eee835/cloudflare/resource_cloudflare_load_balancer_migrate.go#L55-L61

--- a/.changelog/1257.txt
+++ b/.changelog/1257.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/cloudflare_load_balancer: handle empty `rules` for `resourceCloudflareLoadBalancerStateUpgradeV1`
+```

--- a/cloudflare/resource_cloudflare_load_balancer_migrate.go
+++ b/cloudflare/resource_cloudflare_load_balancer_migrate.go
@@ -2,7 +2,6 @@ package cloudflare
 
 import (
 	"context"
-	"reflect"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
@@ -54,7 +53,7 @@ func resourceCloudflareLoadBalancerV0() *schema.Resource {
 }
 
 func resourceCloudflareLoadBalancerStateUpgradeV1(_ context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
-	if !reflect.ValueOf(rawState["rules"]).IsNil() {
+	if rawState["rules"] != nil {
 		for i := range rawState["rules"].([]interface{}) {
 			rawState["rules"].([]interface{})[i].(map[string]interface{})["fixed_response"] = []interface{}{rawState["rules"].([]interface{})[i].(map[string]interface{})["fixed_response"]}
 		}

--- a/cloudflare/resource_cloudflare_load_balancer_migrate.go
+++ b/cloudflare/resource_cloudflare_load_balancer_migrate.go
@@ -2,6 +2,7 @@ package cloudflare
 
 import (
 	"context"
+	"reflect"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
@@ -53,8 +54,10 @@ func resourceCloudflareLoadBalancerV0() *schema.Resource {
 }
 
 func resourceCloudflareLoadBalancerStateUpgradeV1(_ context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
-	for i := range rawState["rules"].([]interface{}) {
-		rawState["rules"].([]interface{})[i].(map[string]interface{})["fixed_response"] = []interface{}{rawState["rules"].([]interface{})[i].(map[string]interface{})["fixed_response"]}
+	if !reflect.ValueOf(rawState["rules"]).IsNil() {
+		for i := range rawState["rules"].([]interface{}) {
+			rawState["rules"].([]interface{})[i].(map[string]interface{})["fixed_response"] = []interface{}{rawState["rules"].([]interface{})[i].(map[string]interface{})["fixed_response"]}
+		}
 	}
 
 	return rawState, nil

